### ISSUE TITLE
How to disable any listener while using emqx docker image

### ISSUE THREAD

**Issue by kiranpradeep** [emqx/emqx-rel#434]
#### Environment

- OS: Ubuntu 18.04.3
- EMQ: 3.2.7

#### Description

During native run (without emqx docker image), I needed to disable a particular listener in emqx (TCP internal listener on port 127.0.0.1:11883). For that, I commented out line starting with`listener.tcp.internal=` in `emqx.conf`. Then emqx started with out TCP internal listener enabled (verified in http dasbboard listeners section).

How do I disable/comment out the same option while starting emqx using docker image (emqx/emqx:v3.2.7) ? 

I tried by providing empty environment variable as (`-e EMQX_LISTENER__TCP__INTERNAL=""`) while giving `docker run` command. But that resulted `Syntax error in /opt/emqx/data/configs/app.xxxxxx.conf after line 886 column 1, parsing incomplete` error by emqx.

**Comment by HJianBo** [comment]
I think the env variables can not comment any lines of conf in the current version.

So the workaround solution is mounting the commented configure file to the  emqx container

**Comment by eisaev** [comment]
I think this issue is similar to https://github.com/emqx/emqx-docker/issues/77
I also faced this issue and solved it locally like this:
```
diff --git a/deploy/docker/docker-entrypoint.sh b/deploy/docker/docker-entrypoint.sh
index 5c0ea2f..bdf9472 100755
--- a/deploy/docker/docker-entrypoint.sh
+++ b/deploy/docker/docker-entrypoint.sh
@@ -121,13 +121,21 @@ do
         # Config in emq.conf
         if [[ ! -z "$(cat $CONFIG |grep -E "^(^|^#*|^#*\s*)$VAR_NAME")" ]]; then
             echo "$VAR_NAME=$(eval echo \$$VAR_FULL_NAME)"
-            echo "$(sed -r "s/(^#*\s*)($VAR_NAME)\s*=\s*(.*)/\2 = $(eval echo \$$VAR_FULL_NAME|sed -e 's/\//\\\//g')/g" $CONFIG)" > $CONFIG   
+            if [[ -z "$(eval echo \$$VAR_FULL_NAME)" ]]; then
+                echo "$(sed -r "s/(^*\s*)($VAR_NAME\s*=\s*.*)/#\2/g" $CONFIG)" > $CONFIG
+            else
+                echo "$(sed -r "s/(^#*\s*)($VAR_NAME)\s*=\s*(.*)/\2 = $(eval echo \$$VAR_FULL_NAME|sed -e 's/\//\\\//g')/g" $CONFIG)" > $CONFIG
+            fi
         fi
         # Config in plugins/*
         for CONFIG_PLUGINS_FILE in $(ls $CONFIG_PLUGINS); do
             if [[ ! -z "$(cat $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE |grep -E "^(^|^#*|^#*\s*)$VAR_NAME")" ]]; then
                 echo "$VAR_NAME=$(eval echo \$$VAR_FULL_NAME)"
-                echo "$(sed -r "s/(^#*\s*)($VAR_NAME)\s*=\s*(.*)/\2 = $(eval echo \$$VAR_FULL_NAME|sed -e 's/\//\\\//g')/g" $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE)" > $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE
+                if [[ -z "$(eval echo \$$VAR_FULL_NAME)" ]]; then
+                    echo "$(sed -r "s/(^*\s*)($VAR_NAME\s*=\s*.*)/#\2/g" $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE)" > $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE
+                else
+                    echo "$(sed -r "s/(^#*\s*)($VAR_NAME)\s*=\s*(.*)/\2 = $(eval echo \$$VAR_FULL_NAME|sed -e 's/\//\\\//g')/g" $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE)" > $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE
+                fi
             fi 
         done
     fi
```
With this patch, a defined environment variable with an empty value is interpreted as a deleted config entry.

**Comment by zhanghongtong** [comment]
Hi @eisaev 
Sorry for the late reply
I think your idea is great, it solves the problem that troubles us, thank you very much

**Comment by zhanghongtong** [comment]
This feature has been implemented in the v4.1-rc.2 version, welcome to use, I will turn off this issue first, if there are any problems, you can always reopen it
